packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -110,13 +110,15 @@ contract TaikoL1 is
         external
         nonReentrant
     {
+        (
+            TaikoData.BlockEvidence memory evidence,
+            TaikoData.BlockMetadata memory metaData
+        ) = abi.decode(
+            input, (TaikoData.BlockEvidence, TaikoData.BlockMetadata)
+        );
         TaikoData.Config memory config = getConfig();
         uint8 maxBlocksToVerify = LibProving.proveBlock(
-            state,
-            config,
-            AddressResolver(this),
-            blockId,
-            abi.decode(input, (TaikoData.BlockEvidence))
+            state, config, AddressResolver(this), blockId, evidence, metaData
```

Comment:
metadata already has an `id` field, so I think we can stop passing `blockId` to LibProving, but we probably want to check `meta.id == blockId`.